src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -411,6 +405,24 @@ public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
         ).collect(Collectors.toList());
     }
 
+    private boolean filterDatasetOnProperties(DatasetDTO dataset, String term) {
+        //datasets need to have consentId, null check to prevent NPE
+        Boolean consentIdMatch;
+        String consentId = dataset.getConsentId();
+        if(Objects.isNull(consentId)) {
+            consentIdMatch = false;
+        } else {
+            consentIdMatch = consentId.toLowerCase().contains(term);
+        }
+        return dataset.getProperties()
+            .stream()
+            .filter(p -> Objects.nonNull(p.getPropertyValue()))
+            .anyMatch(p -> {
+                return consentIdMatch
+                || p.getPropertyValue().toLowerCase().contains(term);
+            });
```

Comment:
Whenever you find yourself creating intermediate holding variables, there is usually a more fluid and less memory-consuming approach. My refactoring approach to this would be to first try and remove the need for `consentId`:

```
        consentIdMatch = Objects.isNull(dataset.getConsentId()) ? 
                false :
                dataset.getConsentId().contains(term);
```

That's a literal translation of 411-416. In general, I like to check the positive case first, then let the negative case be second, so I'd rewrite that as:

```
        consentIdMatch = Objects.nonNull(dataset.getConsentId()) ?
                dataset.getConsentId().contains(term) :
                false;
```
Groovy has a cool operator for this called the [Elvis Operator](https://groovy-lang.org/operators.html#_elvis_operator) which simplifies that even more, but we're in java land

Now that we have a single assignment, we can instead use a primitive, and 7 loc reduces down to 3

```
        boolean consentIdMatch = Objects.nonNull(dataset.getConsentId()) ?
                dataset.getConsentId().contains(term) :
                false;
```

Once I have that, Intelli-j suggests that it can be simplified even more, which it can:

```
        boolean consentIdMatch = Objects.nonNull(dataset.getConsentId()) && 
                dataset.getConsentId().contains(term);
```

Now, we have a single boolean and the properties to look at. By putting the `consentIdMatch` inside the `anyMatch`, we're forcing the code to loop over the properties. In the true case, it will return after the first property, but we don't need to since we know it's a match. We only need to look at properties in the false case. I would take that out of the inside loop and instead move it before the property loop so the look gets short-circuited:

```
        boolean consentIdMatch = Objects.nonNull(dataset.getConsentId()) &&
                dataset.getConsentId().contains(term);
        return consentIdMatch || dataset.getProperties()
                .stream()
                .filter(p -> Objects.nonNull(p.getPropertyValue()))
                .anyMatch(p -> p.getPropertyValue().toLowerCase().contains(term));
```

I think that is as concise as I can imagine it being.